### PR TITLE
Added Xamarin.Mac support

### PR DESCRIPTION
**Pull request by jonlipsky**
* Added support for Xamarin.Mac applications
* Added a Xamarin.Mac sample

Session logging is not implemented for Xamarin.Mac so an additional constant was added (`LIFECYCLE_PROCESSOR`) to for iOS, Android and UWP that do have that functionality.  Previously, `NATIVE_PROCESSOR` was an indication that there were native implementations for both session logging and event processing.

Note 1: The same DSN that was used in the other samples was used.

Note 2: Something similar should be implemented for the Sentry.Maui project for the "net7.0-macos" framework as well.

### CODE REVIEW

**Comment by bruno-garcia** [comment]
I'm fine adding support here fwiw. Just need to get tests to pass

> Note 2: Something similar should be implemented for the Sentry.Maui project for the "net7.0-macos" framework as well.

What part specifically you mean? for MAUI we have already the device info for example. And native crash support even

**Comment by jonlipsky** [comment]
> What part specifically you mean? for MAUI we have already the device info for example. And native crash support even

In Sentry.Maui, you are only building for MacCatalyst, not Mac (AppKit). You're missing a TargetFramework.

image

File: Src/Sentry.Xamarin/Extensions/SentryXamarinOptionsExtensions.cs [hunk 1]
```diff
@@ -112,7 +112,7 @@ internal static bool RegisterNativeIntegrations(this SentryXamarinOptions option
 
         internal static bool RegisterNativeActivityStatus(this SentryXamarinOptions options)
         {
-#if NATIVE_PROCESSOR
+#if LIFECYCLE_PROCESSOR
```
Comment (lucas-zimerman) [inline Src/Sentry.Xamarin/Extensions/SentryXamarinOptionsExtensions.cs:115]: Just wondering why you made this change?

**Comment by lucas-zimerman** [comment]
MacOS is complaining with 
```
  MMP : error MM0074: Xamarin.Mac 8.12.0 does not support a deployment target of 13.1 for macOS (the maximum is 12.3). Please select an older deployment target in your project's Info.plist or upgrade to a newer version of Xamarin.Mac. [/Users/runner/work/sentry-xamarin/sentry-xamarin/Samples/Sample.Xamarin.Mac/Sample.Xamarin.Mac.csproj]
```

File: Src/Sentry.Xamarin/Extensions/SentryXamarinOptionsExtensions.cs (hunk 1 above)
Comment (jonlipsky) [reply Src/Sentry.Xamarin/Extensions/SentryXamarinOptionsExtensions.cs:115]: It's explained in my original pull request comment above:

Session logging is not implemented for Xamarin.Mac so an additional constant was added (LIFECYCLE_PROCESSOR) to for iOS, Android and UWP that do have that functionality. Previously, NATIVE_PROCESSOR was an indication that there were native implementations for both session logging and event processing.

Comment (jonlipsky) [reply Src/Sentry.Xamarin/Extensions/SentryXamarinOptionsExtensions.cs:115]: On the Mac, we don't need to keep track of when the app is backgrounded, or brought to the front (at least in my opinion).

Comment (lucas-zimerman) [reply Src/Sentry.Xamarin/Extensions/SentryXamarinOptionsExtensions.cs:115]: Fair enough

**Comment by bruno-garcia** [comment]
> > What part specifically you mean? for MAUI we have already the device info for example. And native crash support even
> 
> In Sentry.Maui, you are only building for MacCatalyst, not Mac (AppKit). You're missing a TargetFramework.
> 
> image
> `net6-macos` is the missing TargetFramework.

On their docs they mention "through Mac Catalyst": https://learn.microsoft.com/en-us/dotnet/maui/supported-platforms

What's the benefit/goal/tradeoff of supporting also `netx.x-macos` ?
We could continue this on a new issue on the https://github.com/getsentry/sentry-dotnet repo. Mind creating a ticket there please?

**Comment by jonlipsky** [comment]
> What's the benefit/goal/tradeoff of supporting also `netx.x-macos` ?

@bruno-garcia If you want to write a Mac .NET app that uses AppKit, as opposed the Catalyst/iOS API's, then you need to target netX-macos.

MAUI doesn't support an AppKit backend; however like with Xamarin.iOS and Xamarin.Mac, the majority of the code that Sentry would need would be shared, thus it's probably the best place to add it.

**Comment by bruno-garcia** [comment]
> > What's the benefit/goal/tradeoff of supporting also `netx.x-macos` ?
> 
> @bruno-garcia If you want to write a Mac .NET app that uses AppKit, as opposed the Catalyst/UIKit API's, then you need to target netX-macos. (netx-macos, is essentially the latest version of Xamarin.Mac).
> 
> MAUI doesn't support an AppKit backend; however like with Xamarin.iOS and Xamarin.Mac, the majority of the code that Sentry would need would be shared, thus it's probably the best place to add it.

Fair so we're talking about AppKit (Native Mac app with .NET) that targets `netX.Y-macos` which is .NET Core.
I'd be OK adding support on `sentry-dotnet` for it if @bitsandfoxes agrees

**Comment by jonlipsky** [comment]
> Fair so we're talking about AppKit (Native Mac app with .NET) that targets `netX.Y-macos` which is .NET Core. I'd be OK adding support on `sentry-dotnet` for it if @bitsandfoxes agrees

Thanks!

Also keep in mind there are developers, like myself, that do iOS and Android development that don't use MAUI.  For example, my iOS apps target net7-ios; however, they don't reference or use MAUI.

**Review by bitsandfoxes — APPROVED** [review]
Sorry, this took a bit for me to wrap my head around. Thanks for contributing here!

**Comment by bruno-garcia** [comment]
> > Fair so we're talking about AppKit (Native Mac app with .NET) that targets `netX.Y-macos` which is .NET Core. I'd be OK adding support on `sentry-dotnet` for it if @bitsandfoxes agrees
> 
> Thanks!
> 
> Also keep in mind there are developers, like myself, that do iOS and Android development that don't use MAUI. For example, my iOS apps target net7-ios; however, they don't reference or use MAUI.

That's fair. Sentry itself uses [`net7.0-android` for Symbol Collector.](https://github.com/getsentry/symbol-collector/tree/main/src/SymbolCollector.Android) Still benefits of support for Portable PDB to get line numbers in release builds, for example. So we're keen to get the base mobile support in the core SDK. And any MAUI specific only in Sentry.Maui.

**Pull request merged**